### ISSUE TITLE
current bitmap subtitle is not displayed after seeking

### ISSUE THREAD

**Issue by umlx5h** [SuRGeoNix/Flyleaf#491]
Thank you for supporting bitmap subtitles!

I noticed a inconsistent behaviour with internal bitmap subtitles in DVD format compared to SRT subtitles (not bluray).

With bitmap subtitles turned on, if there is already a subtitle playing after you seek, it will not be displayed.
If playback continues after seeking, the next subtitle will be displayed.

there's no problem for SRT subtitles, 
(the bitmap sub is converted to SRT using Subtitle Edit.)

I think it's because the timestamp has exceeded the start of the subtitle when seeking, but I don't have a problem with external subtitles, so could you solve this?

The same video file was checked in mpv and VLC with no problems.

Also, this might be better to make another Issue, but if we increase the `MaxSubsFrames` in Config such as 1 to 5, the subtitles will not be displayed as much as we have increased them after seek.
This was also not a problem with SRT.

**Comment by SuRGeoNix** [comment]
@umlx5h When the subtitles are embedded (even if there are TEXT and not BITMAP) the will not be displayed because the demuxer will seek to the video keyframe while the subtitles are probably a bit before that. So the problem is that with the current design it will not find the nearest subtitle frame. This also related with bad interleaved muxing files.

Similarly for MaxSubsFrames... the external SRT files will be more accurate as the will be seek differently to the exact video frame timstamp.

**Comment by umlx5h** [comment]
I see, so you are saying that the current design is difficult to deal with?
there is no problem with mpv and VLC, so do you have any plans to improve it in the future?

**Comment by SuRGeoNix** [comment]
Yes, sorry for not clarifying that. This is related to this #465.

**Comment by umlx5h** [comment]
I see. Thank you for the explanation.

**Comment by umlx5h** [comment]
I checked again and found the same problem with VLC, I've fixed issue description.

There was no problem with mpv at all.
For mpv, subtitle seek is not possible for bitmap subtitles, but for some reason this problem does not seem to occur.

For example, large jumps using the seek bar do not cause the problem.

It seems that the subtitle information is not cached but the problem does not occur in some way.

https://mpv.io/manual/master/

>  Backward demuxing of subtitles is not supported. Subtitle display still works for some external text subtitle formats. (These are fully read into memory, and only backward display is needed.) Text subtitles that are cached in the subtitle renderer also have a chance to be displayed correctly.

So it may be slightly unrelated to #465.
